dashboard/app/controllers/projects_controller.rb

```diff
@@ -280,6 +280,16 @@ def create_new
     )
   end
 
+  # GET /projects/for_level/:level_id
+  def get_or_create_for_level
+    level = Level.find(params[:level_id])
+    return if redirect_under_13_without_tos_teacher(level)
+    user_storage_id = get_storage_id
+    # find channel for user and level if it exists, or create a new one
+    channel_token = ChannelToken.find_or_create_channel_token(level, request.ip, user_storage_id, nil, {hidden: true})
```

Comment:
Would it be better to just extract the `type` here rather than passing `level` all the way through?

Reply:
The benefits of passing level down are:
- we are already passing level around, so there's no change to the function
- it makes it so channel tokens created via app_options will also have their type set now, as those are already passing a level into the same method